packages/plugin-compiler/src/main/kotlin/io/realm/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -310,4 +404,42 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
             it.dispatchReceiver = irGet(receiver!!)
         }
     }
+
+    private fun IrType.isRealmList(): Boolean {
+        val propertyClassId = this.classifierOrFail.descriptor.classId
+        val realmListClassId = realmListClass.descriptor.classId
+        return propertyClassId == realmListClassId
+    }
+
+    private fun getListGenericType(declaration: IrProperty): List<String> {
+        // Check first if the generic is a subclass of RealmObject
+        val superTypes = declaration.symbol.descriptor.type.arguments[0].type.constructor.supertypes
+        for (superType: KotlinType in superTypes) {
+            if (superType.toString() == "RealmObject") {
+                return listOf("RealmObject")
+            }
+        }
+
+        // Otherwise just return the generic(s) present in the declaration
+        return declaration.symbol.descriptor.type.arguments.map {
+            it.toString()
+        }
+    }
+
+    private fun getCoreTypeFromKotlinType(type: String): String {
```

Comment:
I guess we should consider looking up the actual symbols once and map directly to a compile time element instead of passing these strings around and retrieving the enumeration element every time we are generating the property.

Reply:
This has changed a bit in the upcoming PRs so I will add a note to address it in one of them.